api/app/signals/apps/api/generics/mixins.py

```diff
@@ -6,3 +6,44 @@ def add_user(self, data):
         if request.user and not request.user.is_anonymous:
             data['user'] = request.user.get_username()
         return data
+
+
+class WriteOnceMixin:
```

Comment:
Could this be done at model level?

Reply:
Not sure what you mean. I like this way because we only want to make fields writeonce for the API. In other words if we do this on object level we are not able to fix mistakes using the Django admin (for example). In that case we need to make migrations or queries to fix a typo's in specific fields.